# Sensor identifier
camera: omnis5u3mt9v022:01902a6ff0572e00

# Factory calibration parameters of every camera
# Width and height of images are in pixels
# 2D data matrices recovered using row-wise reconstruction of the data arrays

## THE MANUFACTURER'S CAMERA IDs BELOW ARE NOT THE SAME AS THEIR ROS COUNTERPART ##
## SEE COMMENT BELOW NEXT TO EACH CAMERA IDENTIFIER ##

omni0:  # Corresponding ROS TF: omni0
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.332506,0.154213,-9.5973e-05,-0.000236179,-0.0416498]
  K:
    rows: 3
    cols: 3
    data: [482.047,0,373.237,0,485.211,211.02,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [0.999923,-0.0122837,-0.00147,0.0122886,0.999919,0.00338726,0.00142827,-0.00340507,0.999993]
  T:
    rows: 1
    cols: 3
    data: [0.10017,-4.28285,-56.4338]

omni1:  # Corresponding ROS TF: omni2
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.334384,0.15543,0.000171604,0.000300507,-0.0439626]
  K:
    rows: 3
    cols: 3
    data: [483.259,0,340.948,0,486.027,204.701,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [0.31579,-0.0160136,-0.948694,0.0151516,0.999815,-0.0118331,0.948708,-0.0106374,0.315975]
  T:
    rows: 1
    cols: 3
    data: [2.54238,-5.2387,-64.3326]

omni2:  # Corresponding ROS TF: omni4
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.333605,0.159377,6.11251e-05,4.90177e-05,-0.0460505]
  K:
    rows: 3
    cols: 3
    data: [473.571,0,378.17,0,477.53,212.577,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [-0.806782,0.00919146,-0.590778,0.0121286,0.999926,-0.00100603,0.590725,-0.00797694,-0.806833]
  T:
    rows: 1
    cols: 3
    data: [-2.83123,-13.4152,-54.7062]

omni3:  # Corresponding ROS TF: omni6
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.334747,0.162797,-0.000305541,0.000163014,-0.0517717]
  K:
    rows: 3
    cols: 3
    data: [476.784,0,381.798,0,479.991,205.64,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [-0.807522,-0.0123753,0.589707,-0.0095615,0.999923,0.00789078,-0.58976,0.000733498,-0.807579]
  T:
    rows: 1
    cols: 3
    data: [1.83448,-8.66821,-58.3765]

omni4:  # Corresponding ROS TF: omni8
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.333512,0.157163,-8.2852e-06,0.000265461,-0.0447446]
  K:
    rows: 3
    cols: 3
    data: [478.614,0,377.363,0,481.574,194.839,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [0.310368,0.002664,0.950613,-0.012316,0.999923,0.00121889,-0.950537,-0.012086,0.310377]
  T:
    rows: 1
    cols: 3
    data: [1.30913,-6.09861,-58.8784]

omni5:  # Corresponding ROS TF: omni1
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.334792,0.161382,4.29188e-05,-0.000324466,-0.0476611]
  K:
    rows: 3
    cols: 3
    data: [479.429,0,367.111,0,482.666,230.626,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [0.999985,-0.00444943,-0.00310623,0.00447819,0.999947,0.00931326,0.00306463,-0.00932703,0.999952]
  T:
    rows: 1
    cols: 3
    data: [1.56619,-127.036,-53.1826]

omni6:  # Corresponding ROS TF: omni3
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.337111,0.160611,0.000146382,0.000406074,-0.0464726]
  K:
    rows: 3
    cols: 3
    data: [483.895,0,375.161,0,486.584,220.184,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [0.312653,-0.0185177,-0.949687,0.00414521,0.999827,-0.0181307,0.949858,0.00173195,0.312675]
  T:
    rows: 1
    cols: 3
    data: [3.45397,-128.213,-62.3746]

omni7:  # Corresponding ROS TF: omni5
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.3355,0.162877,4.34759e-05,2.72184e-05,-0.0472616]
  K:
    rows: 3
    cols: 3
    data: [473.368,0,371.65,0,477.558,204.79,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [-0.808777,0.0285286,-0.587423,0.0419269,0.999078,-0.00920491,0.586619,-0.0320735,-0.809228]
  T:
    rows: 1
    cols: 3
    data: [-5.9509,-135.978,-52.6565]

omni8:  # Corresponding ROS TF: omni7
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.348515,0.199388,-0.000381909,8.83314e-05,-0.0801161]
  K:
    rows: 3
    cols: 3
    data: [480.086,0,361.268,0,483.581,221.179,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [-0.807292,-0.022942,0.589706,-0.0215045,0.999724,0.00945438,-0.58976,-0.00504888,-0.807563]
  T:
    rows: 1
    cols: 3
    data: [4.00202,-131.175,-57.9382]

omni9:  # Corresponding ROS TF: omni9
  width: 752
  height: 480
  D:
    rows: 1
    cols: 5
    data: [-0.33305,0.156207,-5.95668e-05,0.000376887,-0.0438085]
  K:
    rows: 3
    cols: 3
    data: [480.918,0,386.897,0,484.086,206.923,0,0,1]
  R:
    rows: 3
    cols: 3
    data: [0.313216,0.00989403,0.94963,-0.0145,0.999879,-0.00563503,-0.949571,-0.0120047,0.313322]
  T:
    rows: 1
    cols: 3
    data: [2.1606,-129.074,-57.785]